lint/quick_fix.py

```diff
@@ -0,0 +1,597 @@
+from collections import defaultdict
+from functools import partial
+from itertools import chain
+import re
+
+import sublime
+import sublime_plugin
+
+from . import persist
+from . import util
+from .generic_text_command import replace_view_content, text_command
+flatten = chain.from_iterable
+
+MYPY = False
+if MYPY:
+    from typing import (
+        Callable, DefaultDict, Dict, Final, Iterable, Iterator, List,
+        NamedTuple, Optional, Set, TypeVar, Union
+    )
+    T = TypeVar("T")
+    S = TypeVar("S")
+    LintError = persist.LintError
+    TextRange = NamedTuple("TextRange", [("text", str), ("range", sublime.Region)])
+    Fixer = Callable[[LintError, sublime.View], Iterator[TextRange]]
+    Fix = Callable[[sublime.View], Iterator[TextRange]]
+
+else:
+    from collections import namedtuple
+    TextRange = namedtuple("TextRange", "text range")
+
+
+class QuickAction:
+    def __init__(self, subject, fn, detail, solves):
+        # type: (str, Fix, Optional[str], List[LintError]) -> None
+        self.subject = subject  # type: Final[str]
+        self.fn = fn  # type: Final[Fix]
+        self.detail = detail  # type: Final[Optional[str]]
+        self.solves = solves  # type: Final[List[LintError]]
+
+    @property
+    def description(self):
+        # type: () -> str
+        return "   ".join(filter(None, (self.subject, self.detail)))
+
+
+class sl_fix_by_ignoring(sublime_plugin.TextCommand):
+    def is_enabled(self, quiet=False):
+        # type: (bool) -> bool
+        if quiet:
+            return len(self.view.sel()) == 1
+        else:
+            return True
+
+    def run(self, edit, quiet=False):
+        # type: (sublime.Edit, bool) -> None
+        view = self.view
+        window = view.window()
+        assert window
+
+        if not quiet:
+            if len(self.view.sel()) != 1:
+                window.status_message("Not implemented for multiple selections")
+                return
+
+        sel = view.sel()[0]
```

Comment:
In stead of requiring 1 selection and having the kinda ugly `quiet` parameter here, would it be feasible to just loop over `view.sel()` and foreach naively? I believe ST prevents having overlapping regions here so there don't seem to be any crazy corner cases to worry about. My first very simple attempt at doing a loop here doesn't work, but what do I know? 😄 

Alternatively, we could still remove the `quiet` parameter. Use `is_visible` in stead of `is_enabled` and simply return `return len(self.view.sel()) == 1` to prevent the command showing up in the palette. And when the command is still called with more than one selection, e.g. via a keybinding where `is_visible` doesn't come into play, we can throw the status message.

Reply:
Actually, `view.sel()[0]` is hardcoded in the code. T.i. the code somewhere assumes there is only one selection. 

Didn't know the `is_visible` thing but I actually don't want to show the quick panel for "No actions." as well.  We try to do this in GitSavvy, and I like it.  For strict key bindings try to flash the status bar with a message to not disturb; on the other hand, if the user is in the panel area, for example using the Command Palette, open a noop panel with an explanation. (Although the `quiet` etxra argument is a hack here, I never needed in GitSavvy.) (An example would be trying to push: if you use a keyboard shortcut, just flash "Already up to date", but if you use the panels display the same info in a noop quick panel.)